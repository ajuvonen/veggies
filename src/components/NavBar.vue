<script setup lang="ts">
import {RouterLink} from 'vue-router';
</script>
<template>
  <nav class="nav">
    <RouterLink to="/log" class="nav__link nav__link--log" :aria-label="$t('views.log')">{{
      $t('general.appTitle')
    }}</RouterLink>
    <div class="flex gap-2">
      <RouterLink class="nav__link" to="/settings" :aria-label="$t('views.settings')">
        <IconComponent icon="cog" size="6vw" class="nav__link-icon" />
      </RouterLink>
    </div>
  </nav>
</template>
<style scoped lang="scss">
.nav {
  @apply w-full;
  @apply flex justify-between items-center;
}

.nav__link {
  @apply rounded-full;
  &:hover {
    @apply text-slate-200 fill-slate-200;
  }
}

.nav__link--log {
  font-family: 'Bungee Shade', system-ui;
  font-size: 6vw;
  @apply uppercase sm:text-4xl rounded-md;
}

.nav__link-icon {
  @apply max-h-9 min-h-5 max-w-9 min-w-5;
}
</style>
