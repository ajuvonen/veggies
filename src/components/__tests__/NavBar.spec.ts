import {describe, it, expect} from 'vitest';
import {mount} from '@vue/test-utils';
import NavBar from '@/components/NavBar.vue';

describe('NavBar', () => {
  it('renders', () => {
    const wrapper = mount(NavBar, {
      props: {
        showStats: false,
      },
    });
    expect(wrapper).toBeTruthy();
  });

  it('uses showStats prop', async () => {
    const wrapper = mount(NavBar, {
      props: {
        showStats: false,
      },
    });
    expect(wrapper.find('[href="/stats"').exists()).toBe(false);
    await wrapper.setProps({showStats: true});
    expect(wrapper.find('[href="/stats"').exists()).toBe(true);
  });
});
