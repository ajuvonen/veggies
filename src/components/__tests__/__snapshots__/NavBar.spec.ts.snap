// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > renders for home 1`] = `
"<nav data-v-c3ceb15a="" class="navbar--home navbar">
  <!--v-if-->
  <div data-v-c3ceb15a="" class="flex-container"><svg style="width: 1.25rem; height: 1.25rem;" class="shrink-0" viewBox="0 0 24 24" aria-hidden="true">
      <path d="M17.9,17.39C17.64,16.59 16.89,16 16,16H15V13A1,1 0 0,0 14,12H8V10H10A1,1 0 0,0 11,9V7H13A2,2 0 0,0 15,5V4.59C17.93,5.77 20,8.64 20,12C20,14.08 19.2,15.97 17.9,17.39M11,19.93C7.05,19.44 4,16.08 4,12C4,11.38 4.08,10.78 4.21,10.21L9,15V16A2,2 0 0,0 11,18M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10,10 0 0,0 12,2Z" class="transition-all"></path>
    </svg><button data-v-24e38f3d="" class="button-like button--text" data-test-id="home-locale-button-fi" aria-label="Change language to Suomi">
      <!--v-if-->fi
    </button></div>
</nav>"
`;

exports[`NavBar > renders for log 1`] = `
"<nav data-v-c3ceb15a="" class="navbar">
  <h1 data-v-c3ceb15a="" aria-label="Track Weekly Progress" class="navbar__header">Eat Your Veggies</h1>
  <div data-v-c3ceb15a="" class="flex-container">
    <!--v-if-->
    <!--v-if--><a data-v-c3ceb15a="" href="/veggies/settings" class="" aria-label="Settings" data-test-id="navbar-settings-link"><svg data-v-c3ceb15a="" style="width: 6vw; height: 6vw;" class="shrink-0 navbar__link-icon" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M12,8A4,4 0 0,1 16,12A4,4 0 0,1 12,16A4,4 0 0,1 8,12A4,4 0 0,1 12,8M12,10A2,2 0 0,0 10,12A2,2 0 0,0 12,14A2,2 0 0,0 14,12A2,2 0 0,0 12,10M10,22C9.75,22 9.54,21.82 9.5,21.58L9.13,18.93C8.5,18.68 7.96,18.34 7.44,17.94L4.95,18.95C4.73,19.03 4.46,18.95 4.34,18.73L2.34,15.27C2.21,15.05 2.27,14.78 2.46,14.63L4.57,12.97L4.5,12L4.57,11L2.46,9.37C2.27,9.22 2.21,8.95 2.34,8.73L4.34,5.27C4.46,5.05 4.73,4.96 4.95,5.05L7.44,6.05C7.96,5.66 8.5,5.32 9.13,5.07L9.5,2.42C9.54,2.18 9.75,2 10,2H14C14.25,2 14.46,2.18 14.5,2.42L14.87,5.07C15.5,5.32 16.04,5.66 16.56,6.05L19.05,5.05C19.27,4.96 19.54,5.05 19.66,5.27L21.66,8.73C21.79,8.95 21.73,9.22 21.54,9.37L19.43,11L19.5,12L19.43,13L21.54,14.63C21.73,14.78 21.79,15.05 21.66,15.27L19.66,18.73C19.54,18.95 19.27,19.04 19.05,18.95L16.56,17.95C16.04,18.34 15.5,18.68 14.87,18.93L14.5,21.58C14.46,21.82 14.25,22 14,22H10M11.25,4L10.88,6.61C9.68,6.86 8.62,7.5 7.85,8.39L5.44,7.35L4.69,8.65L6.8,10.2C6.4,11.37 6.4,12.64 6.8,13.8L4.68,15.36L5.43,16.66L7.86,15.62C8.63,16.5 9.68,17.14 10.87,17.38L11.24,20H12.76L13.13,17.39C14.32,17.14 15.37,16.5 16.14,15.62L18.57,16.66L19.32,15.36L17.2,13.81C17.6,12.64 17.6,11.37 17.2,10.2L19.31,8.65L18.56,7.35L16.15,8.39C15.38,7.5 14.32,6.86 13.12,6.62L12.75,4H11.25Z" class="transition-all"></path>
      </svg></a>
  </div>
</nav>"
`;

exports[`NavBar > renders for settings 1`] = `
"<nav data-v-c3ceb15a="" class="navbar">
  <h1 data-v-c3ceb15a="" aria-label="Settings" class="navbar__header">Settings</h1>
  <div data-v-c3ceb15a="" class="flex-container"><a data-v-c3ceb15a="" href="/veggies/log" class="navbar__log-link" aria-label="Track Weekly Progress" data-test-id="navbar-log-link"><svg data-v-c3ceb15a="" style="width: 6vw; height: 6vw;" class="shrink-0 navbar__link-icon" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.4,16.6L10.8,12L15.4,7.4L14,6L8,12L14,18L15.4,16.6Z" class="transition-all"></path>
      </svg></a>
    <!--v-if--><a data-v-c3ceb15a="" href="/veggies/settings" class="" aria-label="Settings" data-test-id="navbar-settings-link"><svg data-v-c3ceb15a="" style="width: 6vw; height: 6vw;" class="shrink-0 navbar__link-icon" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M12,8A4,4 0 0,1 16,12A4,4 0 0,1 12,16A4,4 0 0,1 8,12A4,4 0 0,1 12,8M12,10A2,2 0 0,0 10,12A2,2 0 0,0 12,14A2,2 0 0,0 14,12A2,2 0 0,0 12,10M10,22C9.75,22 9.54,21.82 9.5,21.58L9.13,18.93C8.5,18.68 7.96,18.34 7.44,17.94L4.95,18.95C4.73,19.03 4.46,18.95 4.34,18.73L2.34,15.27C2.21,15.05 2.27,14.78 2.46,14.63L4.57,12.97L4.5,12L4.57,11L2.46,9.37C2.27,9.22 2.21,8.95 2.34,8.73L4.34,5.27C4.46,5.05 4.73,4.96 4.95,5.05L7.44,6.05C7.96,5.66 8.5,5.32 9.13,5.07L9.5,2.42C9.54,2.18 9.75,2 10,2H14C14.25,2 14.46,2.18 14.5,2.42L14.87,5.07C15.5,5.32 16.04,5.66 16.56,6.05L19.05,5.05C19.27,4.96 19.54,5.05 19.66,5.27L21.66,8.73C21.79,8.95 21.73,9.22 21.54,9.37L19.43,11L19.5,12L19.43,13L21.54,14.63C21.73,14.78 21.79,15.05 21.66,15.27L19.66,18.73C19.54,18.95 19.27,19.04 19.05,18.95L16.56,17.95C16.04,18.34 15.5,18.68 14.87,18.93L14.5,21.58C14.46,21.82 14.25,22 14,22H10M11.25,4L10.88,6.61C9.68,6.86 8.62,7.5 7.85,8.39L5.44,7.35L4.69,8.65L6.8,10.2C6.4,11.37 6.4,12.64 6.8,13.8L4.68,15.36L5.43,16.66L7.86,15.62C8.63,16.5 9.68,17.14 10.87,17.38L11.24,20H12.76L13.13,17.39C14.32,17.14 15.37,16.5 16.14,15.62L18.57,16.66L19.32,15.36L17.2,13.81C17.6,12.64 17.6,11.37 17.2,10.2L19.31,8.65L18.56,7.35L16.15,8.39C15.38,7.5 14.32,6.86 13.12,6.62L12.75,4H11.25Z" class="transition-all"></path>
      </svg></a>
  </div>
</nav>"
`;

exports[`NavBar > renders for stats 1`] = `
"<nav data-v-c3ceb15a="" class="navbar">
  <h1 data-v-c3ceb15a="" aria-label="Statistics" class="navbar__header">Statistics</h1>
  <div data-v-c3ceb15a="" class="flex-container"><a data-v-c3ceb15a="" href="/veggies/log" class="navbar__log-link" aria-label="Track Weekly Progress" data-test-id="navbar-log-link"><svg data-v-c3ceb15a="" style="width: 6vw; height: 6vw;" class="shrink-0 navbar__link-icon" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M15.4,16.6L10.8,12L15.4,7.4L14,6L8,12L14,18L15.4,16.6Z" class="transition-all"></path>
      </svg></a>
    <!--v-if--><a data-v-c3ceb15a="" href="/veggies/settings" class="" aria-label="Settings" data-test-id="navbar-settings-link"><svg data-v-c3ceb15a="" style="width: 6vw; height: 6vw;" class="shrink-0 navbar__link-icon" viewBox="0 0 24 24" aria-hidden="true">
        <path d="M12,8A4,4 0 0,1 16,12A4,4 0 0,1 12,16A4,4 0 0,1 8,12A4,4 0 0,1 12,8M12,10A2,2 0 0,0 10,12A2,2 0 0,0 12,14A2,2 0 0,0 14,12A2,2 0 0,0 12,10M10,22C9.75,22 9.54,21.82 9.5,21.58L9.13,18.93C8.5,18.68 7.96,18.34 7.44,17.94L4.95,18.95C4.73,19.03 4.46,18.95 4.34,18.73L2.34,15.27C2.21,15.05 2.27,14.78 2.46,14.63L4.57,12.97L4.5,12L4.57,11L2.46,9.37C2.27,9.22 2.21,8.95 2.34,8.73L4.34,5.27C4.46,5.05 4.73,4.96 4.95,5.05L7.44,6.05C7.96,5.66 8.5,5.32 9.13,5.07L9.5,2.42C9.54,2.18 9.75,2 10,2H14C14.25,2 14.46,2.18 14.5,2.42L14.87,5.07C15.5,5.32 16.04,5.66 16.56,6.05L19.05,5.05C19.27,4.96 19.54,5.05 19.66,5.27L21.66,8.73C21.79,8.95 21.73,9.22 21.54,9.37L19.43,11L19.5,12L19.43,13L21.54,14.63C21.73,14.78 21.79,15.05 21.66,15.27L19.66,18.73C19.54,18.95 19.27,19.04 19.05,18.95L16.56,17.95C16.04,18.34 15.5,18.68 14.87,18.93L14.5,21.58C14.46,21.82 14.25,22 14,22H10M11.25,4L10.88,6.61C9.68,6.86 8.62,7.5 7.85,8.39L5.44,7.35L4.69,8.65L6.8,10.2C6.4,11.37 6.4,12.64 6.8,13.8L4.68,15.36L5.43,16.66L7.86,15.62C8.63,16.5 9.68,17.14 10.87,17.38L11.24,20H12.76L13.13,17.39C14.32,17.14 15.37,16.5 16.14,15.62L18.57,16.66L19.32,15.36L17.2,13.81C17.6,12.64 17.6,11.37 17.2,10.2L19.31,8.65L18.56,7.35L16.15,8.39C15.38,7.5 14.32,6.86 13.12,6.62L12.75,4H11.25Z" class="transition-all"></path>
      </svg></a>
  </div>
</nav>"
`;
